xds/internal/client/helper.go

```diff
@@ -68,95 +58,90 @@ type Config struct {
 	NodeProto *basepb.Node
 }
 
-// NewConfig returns a new instance of Config initialized by reading the
-// bootstrap file found at ${GRPC_XDS_BOOTSTRAP}. Bootstrap file is read only
-// on the first invocation of this function, and further invocations end up
-// using the results from the former.
-//
-// As of today, the bootstrap file only provides the balancer name and the node
-// proto to be used in calls to the balancer. For transport credentials, the
-// default TLS config with system certs is used. For call credentials, default
-// compute engine credentials are used.
-func NewConfig() (*Config, error) {
-	onceDoerFunc(func() {
-		fName, ok := os.LookupEnv(bootstrapFileEnv)
-		if !ok {
-			bsData, bsErr = nil, fmt.Errorf("xds: %s environment variable not set", bootstrapFileEnv)
-			return
-		}
-		bsData, bsErr = readBootstrapFile(fName)
-	})
-	if bsErr != nil {
-		return nil, bsErr
-	}
-	return &Config{
-		BalancerName: bsData.balancerName(),
-		Creds:        grpc.WithCredentialsBundle(google.NewComputeEngineCredentials()),
-		NodeProto:    bsData.node,
-	}, nil
+type channelCreds struct {
+	Type   string          `json:"type"`
+	Config json.RawMessage `json:"config"`
 }
 
-// bootstrapData wraps the contents of the bootstrap file.
-// Today the bootstrap file contains a Node proto and an ApiConfigSource proto
-// in JSON format.
-type bootstrapData struct {
-	node      *basepb.Node
-	xdsServer *cspb.ApiConfigSource
+type xdsServer struct {
+	ServerURI    string         `json:"server_uri"`
+	ChannelCreds []channelCreds `json:"channel_creds,omitempty"`
```

Comment:
`omitempty` is only useful during encoding, right? Why do we need it here?

Reply:
True. Removed.